rplugin/python3/deoplete/sources/beancount.py

```diff
@@ -0,0 +1,75 @@
+import re
+
+from deoplete.source.base import Base
+
+try:
+    from beancount.loader import load_file
+    from beancount.core.data import Open, Transaction
+    HAS_BEANCOUNT = True
+except ImportError:
+    HAS_BEANCOUNT = False
+
+DIRECTIVES = ['open', 'close', 'commodity', 'txn', 'balance', 'pad', 'note',
+              'document', 'price', 'event', 'query', 'custom']
+
+
+class Source(Base):
+    def __init__(self, vim):
+        super().__init__(vim)
+        self.vim = vim
+
+        self.name = 'beancount'
+        self.mark = '[bc]'
+        self.filetypes = ['beancount']
+        self.min_pattern_length = 0
+
+    def on_init(self, context):
+        if not HAS_BEANCOUNT:
+            self.error('Importing beancount failed.')
+
+    def on_event(self, context):
+        self.__make_cache(context)
+
+    def get_complete_position(self, context):
+        m = re.search(r'\S*$', context['input'])
```

Comment:
Probably this part doesn't matter, but I was always conscious about cursor position during completion.  If you move to the middle of a line and try to complete something, will your code work properly?

Again, I haven't tested it, but I hope to soon.